swaylock/main.c

```diff
@@ -292,6 +294,10 @@ int main(int argc, char **argv) {
 	surfaces = create_list();
 	registry = registry_poll();
 
+	if (!registry) {
+		sway_abort("Unable to connect to sway");
```

Comment:
In this case it's unable to connect to wayland, not sway. And we should probably default to a directory in /tmp instead of just not working if XDG_RUNTIME_DIR isn't set.